### PR TITLE
✨ feat: add coinbase support

### PR DESCRIPTION
**Pull request by insulineru**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -19,7 +21,7 @@ interface IWalletStoreState {
   name: null | AvailableWalletNames
   subName: null | string
   provider: Web3Provider | null
-  walletProvider: WalletConnectProvider | MetaMaskInpageProvider | null
+  walletProvider: WalletConnectProvider | MetaMaskInpageProvider | WalletConnectProvider | CoinbaseWalletProvider | null
```
Comment (insulineru) [inline src/types.ts]:
```suggestion
  walletProvider: WalletConnectProvider | MetaMaskInpageProvider | CoinbaseWalletProvider | null
```

File: src/context/WalletProvider.tsx [hunk 2]
```diff
@@ -16,21 +16,127 @@ import { useBalance } from '../hooks'
...
+
+      localStorage.setItem('web3-wallets-name', WALLET_NAMES.Coinbase)
+      localStorage.setItem(
+        LOCAL_STORAGE_WALLETS_KEY,
+        JSON.stringify({
+          name: WALLET_NAMES.Coinbase,
+          chainId,
+          address: addressDomain || addressShort
+        })
+      )
```
Comment (nrjshka) [inline src/context/WalletProvider.tsx:102]: Use a separate function for this

File: src/context/WalletProvider.tsx [hunk 3]
```diff
@@ -179,12 +286,20 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
 
       localStorage.setItem('web3-wallets-name', WALLET_NAMES.Phantom)
+      localStorage.setItem(
+        LOCAL_STORAGE_WALLETS_KEY,
+        JSON.stringify({
+          name: WALLET_NAMES.Phantom,
+          chainId,
+          address: addressDomain || addressShort
+        })
+      )
```
Comment (nrjshka) [inline src/context/WalletProvider.tsx:302]: Same here

**Review by nrjshka — APPROVED** [review]

File: src/context/WalletProvider.tsx (hunk 2 above)
Comment (insulineru) [reply src/context/WalletProvider.tsx:102]: > Use a separate function for this

I will create an issue with `useLocalStorage`

**Pull request merged**